cogito/put.go

```diff
@@ -160,9 +160,13 @@ func (pu *ProdPutter) ProcessInputDir() error {
 }
 
 func (pu *ProdPutter) Sinks() []Sinker {
-	pu.log.Info("prodPutter: Sinks")
 	return []Sinker{
-		GitHubCommitStatusSink{Putter: pu},
+		GitHubCommitStatusSink{
+			Log:     pu.log,
+			GhAPI:   pu.ghAPI,
+			GitRef:  pu.gitRef,
+			Request: pu.Request,
+		},
```

Comment:
@odormond you remember your comment in previous PR on how to make the sink generic?
This how it is done: the Putter initializes it, setting the fields it needs.
This allows to have, in Put(), a simple and modular `sink.Send() error` without parameters.

Having sink.LoadConfiguration() as you were suggesting would, as I understand it, have two possible implementations:

1. A parameterless LoadConfiguration like Send: `sink.LoadConfiguration() error`. This would require the same mechanism used here in line 164, so no need for LoadConfiguration at all.
2. A LoadConfiguration with parameters. But in this case, how to keep it generic (modular)? Only option would be a io.Reader parameter, and pass to it, _again_, the JSON text to be parsed :-(

I actually though about these options, and this is why, after many different approaches, I selected what we have now.

To be complete, starting with Go 1.18, we could use generics, and so we could pass to sink.Send (or to sink.LoadConfiguration) a type parameter that depends on the implementation (cogito would pass a cogito.PutRequest, another resource foo would pass a foo.PutRequest ...). I tried also this one, but at the end the (slightly) added complexity did not seem justified to me...

(it took a while to come out with the current design, it was not evident!)

Reply:
Thanks for the explanation! I did not consider that the static typing was indeed a problem for a more modular approach.